ISISNeutronMuon/MDANSE
# Update CI/CD on Protos

**Description of work**

The CI/CD has been updated for Protos (Python 3) such that it works in (nearly) its entirety, generating valid installers for all supported platforms. Besides bugs with the GUI (such as some parts like Animation and 2D/3D Plotter either not working or crashing), the generated installers seem to work. Most importantly, the issues described in #147 and #157 do not appear. Issue #159 is difficult to ascertain since Animation does not work on other platforms too.

I have tested on Windows 10, Windows 11, Ubuntu 20, Ubuntu 21, and Ubuntu 22, and MDANSE there installs smoothly except for some issues with Ubuntu 20 and Ubuntu 21, which I am still fixing.

**To test**

1. Download an installer for your platform from the latest push on this branch and install it.
2. Open GUI and see if things generally work (besides bugs like described above)
3. Start the enclosed Python interpreter and check that it works.
